### DISCUSSION TITLE
Default values of Path parameters are missing in OpenAPI doc

### DISCUSSION THREAD

**Discussion by mreiche** [fastapi/fastapi#8475] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/items/{item_id}")
async def read_item(
        item_id: int = Path(42, title="The ID")
):
    return {"item_id": item_id}
```

### Description

I expect the default values are present in the documentation.

<img width="590" alt="image" src="https://user-images.githubusercontent.com/572453/166137315-f87f61a4-2bb6-48d9-92f4-b3cf7d76b567.png">

### Operating System

macOS

### FastAPI Version

 0.75.2

### Python Version

3.9.12

**Comment by Kludex** [comment]
I don't think OpenAPI specs support default value on path... Am I missing it? https://swagger.io/specification/

If it's not there, then there's nothing wrong with FastAPI.

**Comment by mreiche** [comment]
It is supported. I used the petstore and added the default value:
```json
"parameters": [
          {
            "name": "petId",
            "in": "path",
            "description": "ID of pet to update",
            "required": true,
            "type": "integer",
            "format": "int64",
            "default": "cat"
          }
```
and this is the result:
<img width="1513" alt="Bildschirmfoto 2022-05-10 um 17 28 25" src="https://user-images.githubusercontent.com/572453/167666115-aa689632-8a65-43ef-9426-480ae458f4b4.png">

**Reply by nouman-data-scientist** [reply]
@mreiche where did you add this "parameters"?

**Reply by mreiche** [reply]
Here
* https://github.com/swagger-api/swagger-petstore/blob/master/src/main/resources/openapi.yaml
* https://editor.swagger.io/

**Comment by bheuse** [comment]
Same concern - not using Pydantic, cannot set path parameter default value for swagger-ui via fastapi. Pity.

**Comment by mreiche** [comment]
fastapi 0.97.0 prevents `Path` parameters to have default values at all.
```python
path_param = Path("my-default")
```
```
AssertionError: Path parameters cannot have a default value
```
But allows having default parameters for `Query` parameters.
```python
query_param = Query("my-default")
```
The specification doesn't say anything about default values for `PathItems` https://swagger.io/specification/#path-item-object neither `Query` nor `Path`.

That is inconsistent!

**Comment by YuriiMotov** [comment, marked as answer]
> **required**	boolean
> Determines whether this parameter is mandatory. If the [parameter location](https://spec.openapis.org/oas/v3.1.0.html#parameterIn) is "path", this property is REQUIRED and its value MUST be true. Otherwise, the property MAY be included and its default value is false.

Source: https://spec.openapis.org/oas/v3.1.0.html#fixed-fields-9
